addons/website_product_comparison/models/product_comparison.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+from collections import OrderedDict
+
+from openerp import api, fields, models
+
+
+class ProductAttributeCategory(models.Model):
+    _name = "product.attribute.category"
+    _description = "Product Attribute Category"
+    _order = 'sequence'
+
+    name = fields.Char("Category Name", required=True)
+    sequence = fields.Integer("Sequence")
+
+
+class ProductAttribute(models.Model):
+    _inherit = 'product.attribute'
+
+    category_id = fields.Many2one('product.attribute.category', string="Category")
```

Comment:
default+required would be nice but I guess you haven't found a solution for the issue of undefined record at for existing records